<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>SCOPRO</title>


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">SCOPRO</h1>



<p>The vignette depends on CIARA packages.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(SCOPRO)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">required &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;CIARA&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="cf">if</span> (<span class="op">!</span><span class="kw">all</span>(<span class="kw">unlist</span>(<span class="kw">lapply</span>(required, <span class="cf">function</span>(pkg) <span class="kw">requireNamespace</span>(pkg, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)))))</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">  knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">eval =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>In this vignette it is shown the projection performed between single cell RNA seq mouse data from <a href="https://www.nature.com/articles/s41594-021-00590-w">Iturbe et al., 2021</a> and in vivo mouse datasets from <a href="https://pubmed.ncbi.nlm.nih.gov/24408435/">Deng et al. , 2014</a> and <a href="https://www.sciencedirect.com/science/article/pii/S2211124717309610">Mohammed et al. , 2017</a>.</p>
<p>The single cell RNA seq dataset includes 1285 mouse embryonic stem cells, including a small cluster of 2-cell-like cells (2CLC) (cluster 2, 31 cells).</p>
<p>The in vivo mouse dataset from <a href="https://pubmed.ncbi.nlm.nih.gov/24408435/">Deng et al. , 2014</a> includes stages from from early 2 cells-stage to late blastocyst while the in vivo mouse dataset from <a href="https://www.sciencedirect.com/science/article/pii/S2211124717309610">Mohammed et al. , 2017</a> includes stages from from E4.5 to E6.5.</p>
<div id="load-mouse-escs-raw-count-matrix" class="section level2">
<h2>Load mouse ESCs raw count matrix</h2>
<p>We load the raw count matrix provided in the original paper and create norm counts and run cluster analysis with CIARA function <strong>cluster_analysis_integrate_rare </strong></p>
<p>Raw count matrix can be downloaded <a href="https://hmgubox2.helmholtz-muenchen.de/index.php/s/EHQSnjMJxkR7QYT">here</a></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">current_wd &lt;-<span class="st"> </span><span class="kw">getwd</span>()</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">url =<span class="st"> &quot;https://hmgubox2.helmholtz-muenchen.de/index.php/s/EHQSnjMJxkR7QYT/download/SCOPRO.zip&quot;</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">destfile &lt;-<span class="st"> </span><span class="kw">paste0</span>(current_wd,<span class="st">&quot;/SCOPRO.zip&quot;</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">download.file</span>(url, destfile, <span class="dt">quiet =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw">unzip</span>(destfile, <span class="dt">exdir=</span>current_wd)</a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">setwd</span>(<span class="kw">paste0</span>(current_wd,<span class="st">&quot;/SCOPRO&quot;</span>))</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">load</span>(<span class="dt">file=</span><span class="st">'mayra_dati_raw_0.Rda'</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">mayra_seurat_<span class="dv">0</span>=<span class="kw">cluster_analysis_integrate_rare</span>(mayra_dati_raw_<span class="dv">0</span>,<span class="st">&quot;Mayra_data_0&quot;</span>,<span class="fl">0.1</span>,<span class="dv">5</span>,<span class="dv">30</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">norm_es_vitro=<span class="kw">as.matrix</span>(<span class="kw">GetAssayData</span>(mayra_seurat_<span class="dv">0</span>, <span class="dt">slot =</span> <span class="st">&quot;data&quot;</span>,<span class="dt">assay=</span><span class="st">&quot;RNA&quot;</span>))</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">cluster_es_vitro=<span class="kw">as.vector</span>(mayra_seurat_<span class="dv">0</span><span class="op">$</span>RNA_snn_res.<span class="fl">0.1</span>)</a></code></pre></div>
</div>
<div id="load-in-vivo-mouse-datasets" class="section level2">
<h2>Load in vivo mouse datasets</h2>
<p>The seurat object <strong>seurat_genes_published_mouse.Rda</strong> already includes the raw and normalized count matrix obtained combining the two in vivo datasets ( <a href="https://pubmed.ncbi.nlm.nih.gov/24408435/">Deng et al., 014</a> and <a href="https://www.sciencedirect.com/science/article/pii/S2211124717309610">Mohammed et al. , 2017</a> ). Normalization was done with Seurat function <strong>NormalizeData</strong> (default parameters). Seurat object can be downloaded <a href="https://hmgubox2.helmholtz-muenchen.de/index.php/s/EHQSnjMJxkR7QYT">here</a></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">setwd</span>(<span class="kw">paste0</span>(current_wd,<span class="st">&quot;/SCOPRO&quot;</span>))</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">load</span>(<span class="dt">file=</span><span class="st">&quot;seurat_genes_published_mouse.Rda&quot;</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">norm_vivo &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">GetAssayData</span>(seurat_genes_published_mouse, <span class="dt">slot =</span> <span class="st">&quot;data&quot;</span>,<span class="dt">assay=</span><span class="st">&quot;RNA&quot;</span>))</a></code></pre></div>
</div>
<div id="compute-markers-for-selected-in-vivo-stages" class="section level2">
<h2>Compute markers for selected in vivo stages</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw">DefaultAssay</span>(seurat_genes_published_mouse) &lt;-<span class="st"> &quot;RNA&quot;</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4">cluster_mouse_published &lt;-<span class="st"> </span><span class="kw">as.vector</span>(seurat_genes_published_mouse<span class="op">$</span>stim)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"></a>
<a class="sourceLine" id="cb5-7" data-line-number="7">relevant_stages &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Late_2_cell&quot;</span>, <span class="st">&quot;epiblast_4.5&quot;</span>, <span class="st">&quot;epiblast_5.5&quot;</span>, <span class="st">&quot;epiblast_6.5&quot;</span>)</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="kw">DefaultAssay</span>(seurat_genes_published_mouse) &lt;-<span class="st"> &quot;RNA&quot;</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"></a>
<a class="sourceLine" id="cb5-11" data-line-number="11">markers_first_ESC_small &lt;-<span class="st"> </span>CIARA<span class="op">::</span><span class="kw">markers_cluster_seurat</span>(seurat_genes_published_mouse[,cluster_mouse_published<span class="op">%in%</span>relevant_stages],cluster_mouse_published[cluster_mouse_published<span class="op">%in%</span>relevant_stages],<span class="kw">names</span>(seurat_genes_published_mouse<span class="op">$</span>RNA_snn_res.<span class="fl">0.2</span>)[cluster_mouse_published<span class="op">%in%</span>relevant_stages],<span class="dv">10</span>)</a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"></a>
<a class="sourceLine" id="cb6-3" data-line-number="3">markers_mouse &lt;-<span class="st"> </span><span class="kw">as.vector</span>(markers_first_ESC_small[[<span class="dv">3</span>]])</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">stages_markers &lt;-<span class="st"> </span><span class="kw">names</span>(markers_first_ESC_small[[<span class="dv">3</span>]])</a>
<a class="sourceLine" id="cb6-5" data-line-number="5"></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">## Keeping only the genes in common between in vitro and in vivo datasets</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7">stages_markers &lt;-<span class="st"> </span>stages_markers[markers_mouse <span class="op">%in%</span><span class="st"> </span><span class="kw">row.names</span>(norm_es_vitro)]</a>
<a class="sourceLine" id="cb6-8" data-line-number="8"></a>
<a class="sourceLine" id="cb6-9" data-line-number="9">markers_small &lt;-<span class="st"> </span>markers_mouse[markers_mouse <span class="op">%in%</span><span class="st"> </span><span class="kw">row.names</span>(norm_es_vitro)]</a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="kw">names</span>(markers_small) &lt;-<span class="st"> </span>stages_markers</a></code></pre></div>
</div>
<div id="select-only-blackwhite-markers-for-in-vivo-stages" class="section level2">
<h2>Select only black/white markers for in vivo stages</h2>
<p>For each in vivo stage, we select only the markers for which the median is above 0.1 and is below 0.1 in all the other stages.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"></a>
<a class="sourceLine" id="cb7-3" data-line-number="3">marker_result &lt;-<span class="st"> </span><span class="kw">select_top_markers</span>(relevant_stages, cluster_mouse_published, norm_vivo, markers_small, <span class="dt">max_number =</span> <span class="dv">100</span>, <span class="dt">threshold =</span> <span class="fl">0.1</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">marker_all &lt;-<span class="st"> </span>marker_result[[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">marker_stages &lt;-<span class="st"> </span>marker_result[[<span class="dv">2</span>]]</a></code></pre></div>
</div>
<div id="run-scopro" class="section level2">
<h2>Run SCOPRO</h2>
<p>We run SCOPRO between the cluster of the mouse ESCs dataset and the in vivo stage “Late 2-cells”.</p>
<p>The function SCOPRO first computes the mean expression profile of \emph{marker_stages_filter} genes for each cluster in the in vivo and in vitro dataset. For a given cluster, a connectivity matrix is computed with number of rows and number of columns equal to the length of \emph{marker_stages_filter}. Each entry (i,j) in the matrix can be 1 if the fold_change between gene i and gene j is above \emph{fold_change}. Otherwise is 0. Finally the connectivity matrix of Late 2-cells stage and all the clusters in the in vitro dataset are compared. A gene i is considered to be conserved between Late 2-cells stage and an in vitro cluster if the jaccard index of the links of gene i is above .</p>
<p>There are 25 markers of the Late 2-cells stage that are also expressed in the mouse ESC datasets. More than 75% of these 25 markers are conserved in the cluster number 2. This result is expected since cluster 2 is made up by 2CLC, a rare population of cells known to be transcriptionally similar to the late 2 cells-stage in the mouse embryo development (typical markers of 2CLC are the Zscan4 genes, also highly expressed in the late 2 cells-stage).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"></a>
<a class="sourceLine" id="cb8-3" data-line-number="3">marker_stages_filter &lt;-<span class="st"> </span><span class="kw">filter_in_vitro</span>(norm_es_vitro,cluster_es_vitro ,marker_all, <span class="dt">fraction =</span> <span class="fl">0.10</span>, <span class="dt">threshold =</span> <span class="dv">0</span>)</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"></a>
<a class="sourceLine" id="cb8-5" data-line-number="5">analysis_2cell &lt;-<span class="st"> </span><span class="kw">SCOPRO</span>(norm_es_vitro,norm_vivo,cluster_es_vitro,cluster_mouse_published,<span class="st">&quot;Late_2_cell&quot;</span>,marker_stages_filter, <span class="dt">threshold =</span> <span class="fl">0.1</span>, <span class="dt">number_link =</span> <span class="dv">1</span>, <span class="dt">fold_change =</span> <span class="dv">3</span>, <span class="dt">threshold_fold_change =</span> <span class="fl">0.1</span> ,marker_stages, relevant_stages)</a>
<a class="sourceLine" id="cb8-6" data-line-number="6"></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#png(&quot;/Users/gabriele.lubatti/Downloads/SCOPRO_1.png&quot;)</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="kw">plot_score</span>(analysis_2cell, marker_stages, marker_stages_filter, relevant_stages, <span class="st">&quot;Late_2_cell&quot;</span>, <span class="st">&quot;Final score&quot;</span>, <span class="st">&quot;Cluster&quot;</span>, <span class="st">&quot;Late_2_cell&quot;</span>)</a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#dev.off()</span></a></code></pre></div>
</div>
<div id="visualization-of-conserved-not-conserved-genes-between-late-2-cells-stage-and-in-vitro-clusters" class="section level2">
<h2>Visualization of conserved/ not conserved genes between late 2 cells stage and in vitro clusters</h2>
<p>We can visualize which are the markers of the late 2 cells stage that are conserved/ not conserved in cluster 2. As expected the Zscan4 family genes are conserved.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"></a>
<a class="sourceLine" id="cb9-3" data-line-number="3">common_genes &lt;-<span class="st"> </span><span class="kw">select_common_genes</span>(analysis_2cell, marker_stages, relevant_stages, <span class="st">&quot;Late_2_cell&quot;</span>, cluster_es_vitro, <span class="st">&quot;2&quot;</span>)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">no_common_genes &lt;-<span class="st"> </span><span class="kw">select_no_common_genes</span>(analysis_2cell, marker_stages, relevant_stages, <span class="st">&quot;Late_2_cell&quot;</span>, cluster_es_vitro, <span class="st">&quot;2&quot;</span>)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5"></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"></a>
<a class="sourceLine" id="cb9-9" data-line-number="9">all_genes &lt;-<span class="st"> </span><span class="kw">c</span>(no_common_genes[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], common_genes[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</a>
<a class="sourceLine" id="cb9-10" data-line-number="10">all_genes_label &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(no_common_genes[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], <span class="st">&quot;-no_conserved&quot;</span>), <span class="kw">paste0</span>(common_genes[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>], <span class="st">&quot;-conserved&quot;</span>))</a>
<a class="sourceLine" id="cb9-11" data-line-number="11"></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"></a>
<a class="sourceLine" id="cb9-16" data-line-number="16">rabbit_plot &lt;-<span class="st"> </span><span class="kw">plot_score_genes</span>(all_genes, <span class="st">&quot;Mouse ESC&quot;</span>, <span class="st">&quot;Mouse vitro&quot;</span>, norm_es_vitro,norm_vivo[ , cluster_mouse_published<span class="op">==</span><span class="st">&quot;Late_2_cell&quot;</span>],cluster_es_vitro, cluster_mouse_published[cluster_mouse_published <span class="op">==</span><span class="st"> &quot;Late_2_cell&quot;</span>], all_genes_label, <span class="dv">7</span>, <span class="dv">10</span>, <span class="st">&quot;Late_2_cell&quot;</span>)</a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#png(&quot;/Users/gabriele.lubatti/Downloads/SCOPRO_2.png&quot;)</span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18">rabbit_plot</a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#dev.off()</span></a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">utils<span class="op">::</span><span class="kw">sessionInfo</span>()</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt; R version 4.0.2 (2020-06-22)</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt; Running under: macOS Mojave 10.14.6</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">#&gt; [1] C/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-16" data-line-number="16"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"><span class="co">#&gt; [1] SCOPRO_0.1.0</span></a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb10-19" data-line-number="19"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb10-20" data-line-number="20"><span class="co">#&gt;  [1] tidyselect_1.1.1   xfun_0.29          bslib_0.3.1        graphlayouts_0.8.0</span></a>
<a class="sourceLine" id="cb10-21" data-line-number="21"><span class="co">#&gt;  [5] purrr_0.3.4        colorspace_2.0-2   vctrs_0.3.8        generics_0.1.1    </span></a>
<a class="sourceLine" id="cb10-22" data-line-number="22"><span class="co">#&gt;  [9] viridisLite_0.4.0  htmltools_0.5.2    yaml_2.2.1         utf8_1.2.2        </span></a>
<a class="sourceLine" id="cb10-23" data-line-number="23"><span class="co">#&gt; [13] rlang_0.4.12       jquerylib_0.1.4    pillar_1.6.4       glue_1.6.0        </span></a>
<a class="sourceLine" id="cb10-24" data-line-number="24"><span class="co">#&gt; [17] DBI_1.1.2          tweenr_1.0.2       lifecycle_1.0.1    stringr_1.4.0     </span></a>
<a class="sourceLine" id="cb10-25" data-line-number="25"><span class="co">#&gt; [21] munsell_0.5.0      gtable_0.3.0       CIARA_0.1.0        evaluate_0.14     </span></a>
<a class="sourceLine" id="cb10-26" data-line-number="26"><span class="co">#&gt; [25] knitr_1.37         fastmap_1.1.0      parallel_4.0.2     fansi_0.5.0       </span></a>
<a class="sourceLine" id="cb10-27" data-line-number="27"><span class="co">#&gt; [29] tidygraph_1.2.0    Rcpp_1.0.7         scales_1.1.1       jsonlite_1.7.2    </span></a>
<a class="sourceLine" id="cb10-28" data-line-number="28"><span class="co">#&gt; [33] farver_2.1.0       gridExtra_2.3      ggforce_0.3.3      ggplot2_3.3.5     </span></a>
<a class="sourceLine" id="cb10-29" data-line-number="29"><span class="co">#&gt; [37] digest_0.6.29      stringi_1.7.6      ggrepel_0.9.1      dplyr_1.0.7       </span></a>
<a class="sourceLine" id="cb10-30" data-line-number="30"><span class="co">#&gt; [41] polyclip_1.10-0    grid_4.0.2         tools_4.0.2        magrittr_2.0.1    </span></a>
<a class="sourceLine" id="cb10-31" data-line-number="31"><span class="co">#&gt; [45] sass_0.4.0         tibble_3.1.6       ggraph_2.0.5       crayon_1.4.2      </span></a>
<a class="sourceLine" id="cb10-32" data-line-number="32"><span class="co">#&gt; [49] tidyr_1.1.4        pkgconfig_2.0.3    ellipsis_0.3.2     MASS_7.3-52       </span></a>
<a class="sourceLine" id="cb10-33" data-line-number="33"><span class="co">#&gt; [53] viridis_0.6.2      assertthat_0.2.1   rmarkdown_2.11     R6_2.5.1          </span></a>
<a class="sourceLine" id="cb10-34" data-line-number="34"><span class="co">#&gt; [57] igraph_1.2.11      compiler_4.0.2</span></a></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
